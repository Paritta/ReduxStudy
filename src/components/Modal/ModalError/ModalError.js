import React from "react";
import oc from "open-color";
import styled, {keyframes} from "styled-components";
import { fadeIn } from 'react-animations';
import MdClose from "react-icons/lib/md/close";

const propTypes = {
};

const defaultTypes = {
};

const fadeInAnimation = keyframes`${fadeIn}`;

const Wrapper = styled.div`
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    
    border-radius: 10px;
    
    overflow: hidden;
    
    z-index: 10;
    
    width: 550px;
`;

const AnimationWrapper = styled.div`
    animation: 1s ${fadeInAnimation};
    width: 100%;
    
`;

const Dimmed = styled.div`
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
    
    z-index: 5;
    
    background: rgba(0, 0, 0, 0.3);
`;

const ModalBox = styled.div`
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.3);
    
    width: 100%;
    height: 400px;
    
    display: block;
`;

const CloseWrapper = styled.div`
    position: absolute;
    
    top: 50%;
    left: 50%;
    transform: translate(-50%, -70%);
    
    display: flex;
    align-items: center;
    justify-content: center;
    
    border: 5px solid red;
    border-radius: 100%;
    
    height: 40%;
    width: 40%;
`;

const ModalComment = styled.div`
    font-family: 'Hanna', fantasy;
    font-weight: 100;
    font-size: 1.5em;
    
    color: red;
`;

const Button = styled.button`
    background: palevioletred;
    color: white;
    
    font-size: 1em;
    margin: 1em;
    padding: 0.25em 1em;
    
    border: 2px solid palevioletred;
    border-radius: 3px;
    
    transition: all 0.3s ease;
    
    &:hover {
        background: ${oc.pink[4]};
        border: 2px solid ${oc.pink[4]};
    }
`;

const ModalHeader = styled.div`
    font-style: normal;
    font-weight: 400;
    color: white;
    padding: 15px 15px;
    background: palevioletred;
`;

export class ModalError extends React.Component {
    render () {
        return (
            <div>
                <Dimmed />
                <Wrapper>
                    <AnimationWrapper>
                        <ModalHeader>에러</ModalHeader>
                        <ModalBox>
                            <ModalComment>Mesdage</ModalComment>
                            <Button onClick={() => this.hideAnimate()}>나가기</Button>
                        </ModalBox>
                    </AnimationWrapper>
                </Wrapper>
            </div>
        )
    }
}

ModalError.propTypes = propTypes;
ModalError.defaultTypes = defaultTypes;

export default ModalError
