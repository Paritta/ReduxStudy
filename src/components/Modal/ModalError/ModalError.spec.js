import React from "react";
import ReactTestRenderer from "react-test-renderer"
import { ModalError } from "./ModalError";
import "jest-styled-components";

it("ModalError render correctly", () => {
    const tree = ReactTestRenderer.create(
        <ModalError />
    );
    expect(tree.toJSON()).toMatchSnapshot();
});
