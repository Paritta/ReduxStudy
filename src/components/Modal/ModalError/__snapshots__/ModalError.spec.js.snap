// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalError render correctly 1`] = `
.c0 {
  position: fixed;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  z-index: 5;
  background: rgba(0,0,0,0.3);
}

.c1 {
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  border-radius: 10px;
  overflow: hidden;
  z-index: 10;
  width: 550px;
}

.c2 {
  -webkit-animation: 1s eMLfYp;
  animation: 1s eMLfYp;
  width: 100%;
}

.c3 {
  font-style: normal;
  font-weight: 400;
  color: white;
  padding: 15px 15px;
  background: palevioletred;
}

.c4 {
  background: white;
  border: 1px solid rgba(0,0,0,0.3);
  width: 100%;
  height: 400px;
  display: block;
}

.c5 {
  font-family: 'Hanna',fantasy;
  font-weight: 100;
  font-size: 1.5em;
  color: red;
}

.c6 {
  background: palevioletred;
  color: white;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.c6:hover {
  background: #f783ac;
  border: 2px solid #f783ac;
}

<div>
  <div
    className="c0"
  />
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        에러
      </div>
      <div
        className="c4"
      >
        <div
          className="c5"
        >
          Mesdage
        </div>
        <button
          className="c6"
          onClick={[Function]}
        >
          나가기
        </button>
      </div>
    </div>
  </div>
</div>
`;
